### PR TITLE
fix: allow LICENSE rendering via gnoweb

### PR DESCRIPTION
**Pull request by deelawn**
We allow LICENSE files to be included when adding packages, so we should also allow them to be viewed via gnoweb. 

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [x] Provided any useful hints for running manual tests
- [x] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: tm2/pkg/std/memfile.go
Comment (zivkovicmilos) [inline tm2/pkg/std/memfile.go:106]: Can we please transform this if spaghetti into a switch?

File: gno.land/pkg/gnoweb/gnoweb_test.go
Comment (zivkovicmilos) [inline gno.land/pkg/gnoweb/gnoweb_test.go:73]: Why the change?
I'd argue it's not more readable

File: gno.land/pkg/gnoweb/gnoweb.go
Comment (zivkovicmilos) [inline gno.land/pkg/gnoweb/gnoweb.go:117]: What do you think about extracting these regex expressions?

This way, we can easily test them out and verify they work

File: tm2/pkg/std/memfile.go
Comment (zivkovicmilos) [inline tm2/pkg/std/memfile.go:93]: Aren't documents like `LICENSE.md` or `LICENSE.txt` also valid license documents, regardless of their file extension?
Why do we force them to be named exclusively `LICENSE`?

**Review by zivkovicmilos — APPROVED** [review]
I've left some comments that should be addressed before merging, otherwise looks good 💯

File: tm2/pkg/std/memfile.go
Comment (deelawn) [reply tm2/pkg/std/memfile.go]: Yes, done

File: gno.land/pkg/gnoweb/gnoweb_test.go
Comment (deelawn) [reply gno.land/pkg/gnoweb/gnoweb_test.go]: Didn't mean to commit that; used for debugging. Reverted

File: tm2/pkg/std/memfile.go
Comment (deelawn) [reply tm2/pkg/std/memfile.go:93]: This definitely warrants a larger discussion, but it is like this currently because this is the "allowed" filename for licenses:
https://github.com/gnolang/gno/blob/31a5f2ea63ab1f069201bfa7ad35e9eada8ec6d9/gnovm/pkg/gnolang/nodes.go#L1110-L1114

There are some other parts to this that seems strange and need to be better considered so I created an issue:
https://github.com/gnolang/gno/issues/2421

The particular issue that this PR addresses will unblock me on something; I can fill you in on the details.

File: gno.land/pkg/gnoweb/gnoweb.go
Comment (deelawn) [reply gno.land/pkg/gnoweb/gnoweb.go:117]: I think it makes sense for this particular expression to addressed as a part of the fix for #2421, when the logic becomes more unified.

**Review by leohhhn — APPROVED** [review]
Just check TM2 lint in CI, otherwise seems good.

**Pull request merged**
